examples/record.zig

```diff
@@ -44,20 +44,18 @@ pub fn main() !void {
     //
     // ```
     // paplay -p --format=FLOAT32LE --rate 48000 --raw zig-out/raw_audio
-    // aplay -c 2 -f FLOAT_LE -r 48000 zig-out/raw_audio
+    // aplay -f FLOAT_LE -r 48000 zig-out/raw_audio
     // ```
 
     while (true) {}
 }
 
-fn readCallback(_: ?*anyopaque, frames: usize) void {
-    var bw = std.io.bufferedWriter(file.writer());
-    for (0..frames) |fi| {
-        for (recorder.channels()) |ch| {
-            const sample = recorder.read(ch, fi, f32);
-            const sample_bytes: [4]u8 = @bitCast(sample);
-            _ = bw.write(&sample_bytes) catch {};
-        }
-    }
-    bw.flush() catch {};
+fn readCallback(_: ?*anyopaque, input: []const u8) void {
+    const format_size = recorder.format().size();
+    const frames = input.len / format_size;
+
+    var buffer: [16 * 1024]f32 = undefined;
+    recorder.read(f32, buffer[0..frames], input);
```

Comment:
Is there any reason that `recorder.read` (and `.write`) need to be methods? This API design feels very strange:

* Your `readCallback` is called with the `input` from the microphone
* You then need to ask the `recorder` to `read` data, ??? what does that even mean, weren't we given it as `input`?

I think `sysaudio.convert` would probably be a better way to phrase `read`, I have some more thoughts I'll type up.

Reply:
My thoughts: https://github.com/hexops/mach/issues/1099

Reply:
fair enough. the name is indeed confusing. but we can't make convert them into a single `convert` function because it requires unneeded ugly casts as both `input` and `output` has to be `u8` slices.